src/editor/parser/marked.js

```diff
@@ -331,13 +331,19 @@ Lexer.prototype.token = function(src, top, bq) {
           }
         }
 
+        var prevItem = '';
+        if (i === 0) {
+          prevItem = item;
+        } else {
+          prevItem = cap[i - 1];
+        }
+
         // Determine whether item is loose or not.
-        // Use: /(^|\n)(?! )[^\n]+\n\n(?!\s*$)/
-        // for discount behavior.
-        loose = next || /\n\n(?!\s*$)/.test(item);
-        if (i !== l - 1) {
-          next = item.charAt(item.length - 1) === '\n';
-          if (!loose) loose = next;
+        loose = /^ *([*+-]|\d+\.) +\S+\n\n(?!\s*$)/.test(itemWithBullet);
```

Comment:
I am a little confused,  why remove the variable `next` ?

Reply:
I forgot to change/update this. I just updated it :)